python/lsst/pipe/base/cmdLineTask.py

```diff
@@ -464,10 +465,19 @@ def parseAndRun(cls, args=None, config=None, log=None, doReturnResults=False):
             This will typically be a list of `None` unless doReturnResults is `True`;
             see cls.RunnerClass (TaskRunner by default) for more information.
         """
+        if args is None:
+            commandAsStr = " ".join(sys.argv)
+            args = sys.argv[1:]
+        else:
+            commandAsStr = "{}{}".format(lsst.utils.get_caller_name(skip=1), tuple(args))
+
         argumentParser = cls._makeArgumentParser()
         if config is None:
             config = cls.ConfigClass()
         parsedCmd = argumentParser.parse_args(config=config, args=args, log=log, override=cls.applyOverrides)
+        # print this message after parsing the command so the log is fully configured
+        parsedCmd.log.info("Running: {}".format(commandAsStr))
```

Comment:
My understanding is that delayed formatting is useful primarily when a log message may not be displayed. I would expect info and fatal to always be displayed. Still...it ought to be a harmless change and I guess it is good log style.